protostar/testing/test_collector.py

```diff
@@ -275,12 +277,25 @@ def _build_test_suite_from_test_suite_info(
         test_suite_info: TestSuiteInfo,
     ) -> TestSuite:
         function_names = self._get_suite_function_names(test_suite_info.path)
-        setup_fn_name = self._collect_setup_hook_name(function_names)
+
+        if len(function_names) == 0:
+            names: List[str] = []
+            gas: List[AvailableGas] = []
+        elif isinstance(function_names[0], str):
+            function_names = cast(List[str], function_names)
+            names: List[str] = function_names
+            gas: List[AvailableGas] = [None] * len(function_names)
+        else:
+            function_names = cast(List[Tuple[str, AvailableGas]], function_names)
+            names: List[str] = [fn[0] for fn in function_names]
+            gas: List[AvailableGas] = [fn[1] for fn in function_names]
+
+        setup_fn_name = self._collect_setup_hook_name(names)
 
         test_cases = list(
             test_suite_info.filter_test_cases(
                 self._collect_test_cases(
-                    function_names=[(fn, None) for fn in function_names],
+                    function_names=list(zip(names, gas)),
```

Comment:
This `zip` is not needed for the `else` case, right? Because you already have them zipped. If I understood correctly. If so, then, for the `else` case what you do is first unzip and zip it back, right? In such a case I think it would be better to zip only for the `elif` case and for the `else` case just assign to the `zip` the value you already have.

Reply:
This function has ~20 lines, and having one reference to a variable will not ruin readability. I guess you probably will not be convinced by whatever I say, @Arcticae can you share your opinion so we can move on?

Reply:
I think what's happenning actually is that this function does two things depending on the type and that's what causing the problems. As for the style really it's not that big of an issue, i like the code as-is but i can see your point @karol-bisztyga